app/controllers/users_controller.rb

```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def show
+    @recipe_foods = RecipeFood.joins(:food, :recipe).where(food: { user: current_user }, recipe: { user: current_user })
+    @items_count = @recipe_foods.select('food_id').distinct.count
+    @total_amount = @recipe_foods.sum('quantity * price')
+    @items = @recipe_foods.group('food.name, price, measurement_unit')
+      .select('food.name, SUM(quantity) as quantity, (sum(quantity) * price) as price, measurement_unit')
+  end
```

Comment:
- [ ] We don't implement SQL Queries inside the controller, we use the model to keep the SQL queries implementation.